client/extensions/woocommerce/state/sites/settings/mailchimp/test/actions.js

```diff
@@ -347,8 +347,85 @@ describe( 'actions', () => {
 	describe( '#mailChimpSaveSettings()', () => {
 		const siteId = '123';
 
-		test( 'should dispatch an action', () => {
+		test( 'should not dispatch an action if ther are not settings in state', () => {
 			const getState = () => ( {} );
+			const dispatch = spy();
+			mailChimpSaveSettings( siteId )( dispatch, getState );
+			expect( dispatch ).to.have.not.been.calledWith( {
+				type: WOOCOMMERCE_MAILCHIMP_SAVE_SETTINGS,
+				siteId,
+			} );
+		} );
+
+		test( 'should not dispatch an action if the state does not represent compleated setup', () => {
+			const getState = () => ( {
+				extensions: {
+					woocommerce: {
+						sites: {
+							[ siteId ]: {
+								settings: {
+									mailchimp: {
+										settings: {
+											mailchimp_api_key: '12345testing',
+											mailchimp_debugging: false,
+											mailchimp_account_info_id: '64c533c4',
+											mailchimp_account_info_username: 'ski',
+											mailchimp_active_tab: 'campaign_defaults',
```

Comment:
here you have `mailchimp_active_tab` but above you are testing `active_tab`

Reply:
Yes that is true. Here `active_tab` is just missing for some reason - this means that MailChimp plugin is not configured yet. Compare that to my answer to your comment below  https://github.com/Automattic/wp-calypso/pull/20866/files/147b6d26d4d89b4462f727ee98157200c472d3a1#r157561885